hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java

```diff
@@ -180,39 +185,53 @@ public ExpirationManager(PartitionContainer[] partitionContainers, NodeEngine no
      * Starts scheduling of the task that clears expired entries.
      * Calling this method multiple times has same effect.
      */
-    private synchronized void start() {
-        if (expirationTask != null) {
+    public void scheduleExpirationTask() {
+        if (expirationTask != null
+                || nodeEngine.getLocalMember().isLiteMember()) {
             return;
         }
 
-        ClearExpiredRecordsTask task = new ClearExpiredRecordsTask();
-        expirationTask = executionService.getGlobalTaskScheduler().
-                scheduleWithRepetition(task, taskPeriodSeconds, taskPeriodSeconds, SECONDS);
+        synchronized (this) {
+            if (expirationTask != null) {
+                return;
+            }
+
+            ClearExpiredRecordsTask task = new ClearExpiredRecordsTask();
+            expirationTask = executionService.getGlobalTaskScheduler().
+                    scheduleWithRepetition(task, taskPeriodSeconds, taskPeriodSeconds, SECONDS);
+
+            expirationTaskEverScheduled.set(true);
+        }
     }
 
     /**
      * Ends scheduling of the task that clears expired entries.
      * Calling this method multiple times has same effect.
      */
-    private synchronized void stop() {
+    private void unscheduleExpirationTask() {
         if (expirationTask == null) {
             return;
         }
 
-        expirationTask.cancel(true);
-        expirationTask = null;
+        synchronized (this) {
+            if (expirationTask == null) {
+                return;
+            }
+
+            expirationTask.cancel(true);
```

Comment:
Underlying `executionService` has multiple executing threads, so seems like it's possible to have multiple expiration tasks running in parallel on the same partitions, since `cancel` doesn't abort the already running task if there is one. It's ok to have overlapping tasks for some short period of time if all involved structures support concurrent access, but do they really support it?

Reply:
Seems possible, added a check for that